rero_ils/modules/items/listener.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of RERO ILS.
+# Copyright (C) 2017 RERO.
+#
+# RERO ILS is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# RERO ILS is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RERO ILS; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, RERO does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""."""
+
+from .api import ItemsSearch
+
+
+def enrich_item_data(sender, json=None, record=None, index=None,
+                     **dummy_kwargs):
+    """Signal sent before a record is indexed.
+
+    Arguments:
+    - ``json``: The dumped record dictionary which can be modified.
+    - ``record``: The record being indexed.
+    - ``index``: The index in which the record will be indexed.
+    - ``doc_type``: The doc_type for the record.
+    """
+    # TODO: this multiply the indexing time by 5, try an other way!
+    item_index_name = ItemsSearch.Meta.index
+    if index.startswith(item_index_name):
+        org_pid = record.get_library().replace_refs()['organisation']['pid']
```

Comment:
This line fails when running `setup.sh`:
```
  File "/Users/baa/devel/rero/rero-ils/rero_ils/modules/items/listener.py", line 43, in enrich_item_data
    org_pid = record.get_library().replace_refs()['organisation']['pid']
AttributeError: 'Record' object has no attribute 'get_library'
```

Reply:
Solved